subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/IndexedNormalizedFileSnapshot.java

```diff
@@ -61,4 +62,25 @@ protected int hashNormalizedPath() {
         }
         return h;
     }
+
+    @Override
+    protected int compareNormalizedPathTo(NormalizedFileSnapshot other) {
```

Comment:
❌ I guess some unit tests for comparing indexed normalized paths would be nice. Do we have the coverage already somewhere?

Reply:
We don't have direct coverage yet, only some indirect failures when you mess this one up https://builds.gradle.org/viewLog.html?buildId=12299634&tab=buildResultsDiv&buildTypeId=Gradle_Check_Stage_BranchBuildAccept_Trigger&logTab=

I'll add some direct tests.